staging/src/k8s.io/component-base/logs/logs.go

```diff
@@ -54,8 +105,19 @@ func (writer KlogWriter) Write(data []byte) (n int, err error) {
 func InitLogs() {
 	log.SetOutput(KlogWriter{})
 	log.SetFlags(0)
-	// The default glog flush interval is 5 seconds.
-	go wait.Forever(klog.Flush, *logFlushFreq)
+	go flushForever()
+}
+
+func flushForever() {
```

Comment:
how does this differ from wait.Forever?

Reply:
`wait.Forever` is passed the period once and value changes don't have an effect. This new code looks at the current period each time it needs to decide how long to sleep, therefore changes in that value have an effect.

Reply:
Oh, now I get what you are doing.  Can you add something to that effect to the description of flushForever?

For my knowledge, why/how would someone re-configure a flag?  Aren't flags only set once?

Reply:
Usually yes, although that isn't required. The same flag set could also be used multiple times. The main reason for this change is that the logging is initialized in main() (which makes sense - the sooner the better) and flag parsing happens at a later time when the goroutine is already running.

I've added a comment explaining this.

Reply:
Should we just require InitLogs to be called after flag parsing?  We can recommend calling `defer logs.FlushLogs()`  in main() ASAP to capture logs for cases where logging isn't setup yet.

Reply:
I mean... I lean towards the flush frequency not being configurable after startup.  How many places would we have to change?

Reply:
I just counted 19, in various parts of the code base. Simply from a practical point of view I don't look forward to modify all of those.

But it's not just that. I think the current API with `InitLogs` being called as early as possible makes sense. It's easy to use and review in commands ("InitLogs and FlushLogs called in main()? Okay, done."). In the package, it gives us an opportunity to make some changes before log messages are emitted.